src/hyperloglog.c

```diff
@@ -1041,6 +1045,120 @@ int hllAdd(robj *o, unsigned char *ele, size_t elesize) {
     }
 }
 
+#if defined(HAVE_AVX2)
+/* Optimized for the default configuration: 16384 registers of 6 bits each. */
+ATTRIBUTE_TARGET_AVX2
+void hllMergeDense_AVX2(uint8_t *reg_raw, const uint8_t *reg_dense) {
+    const __m256i shuffle = _mm256_setr_epi8( //
+        4, 5, 6, -1,                             //
+        7, 8, 9, -1,                             //
+        10, 11, 12, -1,                          //
+        13, 14, 15, -1,                          //
+        0, 1, 2, -1,                             //
+        3, 4, 5, -1,                             //
+        6, 7, 8, -1,                             //
+        9, 10, 11, -1                            //
+    );
+
+    /* Merge the first 32 registers normally 
+     * as the AVX2 algorithm needs 4 padding bytes at the start */
+    uint8_t val;
+    for (int i = 0; i < 32; i++) {
+        HLL_DENSE_GET_REGISTER(val, reg_dense, i);
+        if (val > reg_raw[i]) {
+            reg_raw[i] = val;
+        }
+    }
+
+    /* Dense to Raw:
+     *
+     * 4 registers in 3 bytes:
+     * {bbaaaaaa|ccccbbbb|ddddddcc}
+     * 
+     * LOAD 32 bytes (32 registers) per iteration:
+     * 4(padding) + 12(16 registers) + 12(16 registers) + 4(padding)
+     * {XXXX|AAAB|BBCC|CDDD|EEEF|FFGG|GHHH|XXXX}
+     * 
+     * SHUFFLE to:
+     * {AAA0|BBB0|CCC0|DDD0|EEE0|FFF0|GGG0|HHH0}
+     * {bbaaaaaa|ccccbbbb|ddddddcc|00000000} x8
+     *
+     * extract registers in each group with AND and SHIFT:
+     * {00aaaaaa|00000000|00000000|00000000} x8
+     * {00000000|00bbbbbb|00000000|00000000} x8
+     * {00000000|00000000|00cccccc|00000000} x8
+     * {00000000|00000000|00000000|00dddddd} x8
+     *
+     * merge the extracted registers with OR:
+     * {00aaaaaa|00bbbbbb|00cccccc|00dddddd} x8
+     *
+     * Finally, compute MAX(reg_raw, merged) and STORE it back to reg_raw
+     */
+
+    const uint8_t *r = reg_dense + 24 - 4;
+    uint8_t *t = reg_raw + 32;
+
+    for (int i = 0; i < HLL_REGISTERS / 32 - 2; ++i) {
```

Comment:
```suggestion
    const uint8_t *r = reg_dense + 6 - 4;
    uint8_t *t = reg_raw + 8;

    for (int i = 0; i < HLL_REGISTERS / 32 - 1; ++i) {
```

so that we can reduce 32 calls of fallback method, and tcl tests also are passed, i also did performance tests by your scripts, there is a little improvement, and just test once, please take a look

```
current: Pfcounts     79993.65         2.49851         2.46300         4.92700         5.82300      4296.53
cr     : Pfcounts     80441.25         2.48559         2.43100         4.86300         6.14300      4320.57

current: Pfmerges    148759.06         1.34511         1.31900         2.63900         4.25500      9297.44
cr     : Pfmerges    152308.85         1.31122         1.28700         2.57500         4.25500      9519.30
```